src/components/Post/Comment/CommentDisplay.tsx

```diff
@@ -9,18 +11,23 @@ const CommentDisplay = ({comment}: {comment: Comment}) => {
     const [author, setAuthor] = useState<User>();
     const [body, setBody] = useState('');
     const [postTime, setPostTime] = useState<string>();
+    const { isLoading, error, data } = useQuery(comment.docRef!.id, buildCommentFetcher(comment));
 
     useEffect( () => {
-        comment?.getTextContent().then( (text) => setBody(text));
-        comment?.getAuthor().then(setAuthor);
-        // comment?.getTimestamp().then( (val) => {
-        // })
-    }, [comment]);
+        if (data)
```

Comment:
We should explicitly check undefined, theoretically we could write a fetcher that returns 'false'

Reply:
Sounds good